cores/esp8266/time.cpp

```diff
@@ -208,6 +208,19 @@ void configTime(int timezone_sec, int daylightOffset_sec, const char* server1, c
     sntp_init();
 }
 
+void configTime(int timezone_sec, int daylightOffset_sec, String server1, String server2, String server3)
+{
+    static String servers[3];
+    servers[0] = std::move(server1);
+    servers[1] = std::move(server2);
+    servers[2] = std::move(server3);
+
```

Comment:
I suspect there is one copy and one move
- copy: at user calling level
- move: here

It can be opposed to this case where there'd be only one copy:
```cpp
void configTime(int timezone_sec, int daylightOffset_sec, const String& server1, const String& server2, const String& server3)
{
    servers[0] = server1; ...
```


This is more of a question than anything else.
I was used to using `std::move()` only with rvalue references which prevent from passing temporaries (read: my understanding stopped there).
```cpp
void configTime(int timezone_sec, int daylightOffset_sec, String&& server1, String&& server2, String&& server3)
{
    servers[0] = std::move(server1); ...
```
but in this case, user string would be cleared/zeroed after this call.

Reply:
depends on how it's called. here it is rvalue from the `String()` pov, so it is moved all the way
```cpp
String getSomeServer();

void configure() {
  configTime("UTC0", getSomeServer());
}
```

this will be a const ref, meaning a copy plus move
```cpp
String blah("best.server.ntp.org");
configTime("UTC0", best);
```

`std::move()` is just a `&&` cast on the original type. in this case, we force a `String::operator=(String&&)` overload, otherwise it would use `String::operator=(const String&)`
since we are making a copy anyway, might as well avoid writing `const String&` and `String&&` overloads